### PR TITLE
Kubernetes Integration Test Cleanup

### PR DESCRIPTION
**Pull request by johnbelamaric**
Improved the testdata struct to include Rcode and actual RRs.
Now checks Rcode not just counts. Also checks counts. Checking
actual RR values still TODO.

### CODE REVIEW

File: test/kubernetes_test.go [hunk 1]
```diff
@@ -12,63 +11,158 @@ import (
...
-	{"mynginx.*.svc.coredns.local.", 1, 1},                       // One A record, via wildcard namespace
-	{"mynginx.any.svc.coredns.local.", 1, 1},                     // One A record, via wildcard namespace
-	{"someservicethatdoesnotexist.*.svc.coredns.local.", 0, 0},   // Record does not exist with wildcard for namespace
-	{"someservicethatdoesnotexist.any.svc.coredns.local.", 0, 0}, // Record does not exist with wildcard for namespace
-	{"*.demo.svc.coredns.local.", 2, 2},                          // Two A records, via wildcard
-	{"any.demo.svc.coredns.local.", 2, 2},                        // Two A records, via wildcard
-	{"*.test.svc.coredns.local.", 0, 0},                          // Two A record, via wildcard that is not exposed
-	{"any.test.svc.coredns.local.", 0, 0},                        // Two A record, via wildcard that is not exposed
-	{"*.*.svc.coredns.local.", 2, 2},                             // Two A records, via namespace and service wildcard
+func makeRR(s string) dns.RR {
```
Comment (miekg) [inline test/kubernetes_test.go]: test.A and test.MX our your friends here - at least they are used in other tests.

File: test/kubernetes_test.go [hunk 2]
```diff
@@ -12,63 +11,158 @@ import (
...
+	answer testanswer
+}
 
-	// t.Skip("Skip Kubernetes Integration tests")
-	// subtests here (Go 1.7 feature).
-	testLookupA(t)
-	testLookupSRV(t)
+// Test data
+// TODO: Fix the actual RR values
+var testdata = []testdefn{
```
Comment (miekg) [inline test/kubernetes_test.go]: There is also infra for this - look at the other tests (can remember from the top of my head which one)

**Review by miekg — CHANGES_REQUESTED** [review]
Welcome this change, but please use the code that already exists.

File: test/kubernetes_test.go (hunk 2 above)
Comment (johnbelamaric) [reply test/kubernetes_test.go]: Ack. See that now. Should have checked before I did it!

File: test/kubernetes_test.go [hunk 3]
```diff
@@ -3,72 +3,169 @@
 package test
 
 import (
-	"io/ioutil"
-	"log"
 	"testing"
 	"time"
 
 	"github.com/mholt/caddy"
+	"github.com/miekg/coredns/middleware/test"
```
Comment (miekg) [inline test/kubernetes_test.go:10]: Nit; could you separate these out in coredns import and the rest? So stdlib, coredns, rest?

File: test/kubernetes_test.go [hunk 4]
```diff
@@ -3,72 +3,169 @@
...
-	testLookupSRV(t)
+// Test data
+// TODO: Fix the actual RR values
+
+var dnsTestCases = []test.Case{
+	{
+		Qname: "mynginx.demo.svc.coredns.local.", Qtype: dns.TypeA,
+		Rcode: dns.RcodeSuccess,
+		Answer: []dns.RR{
+			test.A("mynginx.demo.svc.coredns.local.      1800    IN      A       10.3.0.10"),
```
Comment (miekg) [inline test/kubernetes_test.go:22]: For future reference ('cause it is hard to find): a TTL of 303 means: don't check ttl

File: test/kubernetes_test.go [hunk 5]
```diff
@@ -85,96 +182,43 @@ func createTestServer(t *testing.T, corefile string) (*caddy.Instance, string) {
...
+			t.Errorf("Expected %d answers but got %d for query %s, %d", len(tc.Answer), len(res.Answer), tc.Qname, tc.Qtype)
 		}
 
-		if srvRecordCount != testData.SRVRecordCount {
-			t.Errorf("Expected '%v' SRV records in response. Instead got '%v' SRV records. Test query string: '%v', res: %v", testData.SRVRecordCount, srvRecordCount, testData.Query, res)
-		}
-		if len(res.Answer) != testData.TotalAnswerCount {
-			t.Errorf("Expected '%v' records in answer section. Instead got '%v' records in answer section. Test query string: '%v', res: %v", testData.TotalAnswerCount, len(res.Answer), testData.Query, res)
-		}
+		//TODO: Check the actual RR values
```
Comment (miekg) [inline test/kubernetes_test.go:222]: See `test.Section` and `test.Header`

**Comment by johnbelamaric** [comment]
@chrisohaver already pulled these in when fixing #422 .

**Pull request closed without merging**